plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java

```diff
@@ -1596,5 +1626,151 @@ private static void registerJdkInternalMiscUnsafeIntrinsics(final CompilationCon
         };
 
         intrinsics.registerIntrinsic(unsafeDesc, "unalignedAccess0", emptyToBool, unalignedAccess0);
+
+        InstanceIntrinsic objectFieldOffset1 = (builder, instance, target, arguments) -> {
+            Value clazz = traverseLoads(arguments.get(0));
+            Value string = traverseLoads(arguments.get(1));
+            LiteralFactory lf = ctxt.getLiteralFactory();
+            ObjectType objectType;
+            if (clazz instanceof ClassOf) {
+                ClassOf classOf = (ClassOf) clazz;
+                Value typeId = classOf.getInput();
+                if (typeId instanceof TypeLiteral) {
+                    ValueType valueType = ((TypeLiteral) typeId).getValue();
+                    if (valueType instanceof ObjectType) {
+                        objectType = (ObjectType) valueType;
+                    } else {
+                        ctxt.error(builder.getLocation(), "objectFieldOffset type argument must be a literal of an object type");
+                        return lf.literalOf(0L);
+                    }
+                } else {
+                    ctxt.error(builder.getLocation(), "objectFieldOffset type argument must be a literal of an object type");
+                    return lf.literalOf(0L);
+                }
+            } else {
+                ctxt.error(builder.getLocation(), "objectFieldOffset type argument must be a literal of an object type");
+                return lf.literalOf(0L);
+            }
+            String fieldName;
+            if (string instanceof StringLiteral) {
+                fieldName = ((StringLiteral) string).getValue();
+            } else if (string instanceof ObjectLiteral) {
+                VmObject vmObject = ((ObjectLiteral) string).getValue();
+                if (! (vmObject instanceof VmString)) {
+                    ctxt.error(builder.getLocation(), "objectFieldOffset string argument must be a literal string");
+                    return lf.literalOf(0L);
+                }
+                fieldName = ((VmString) vmObject).getContent();
+            } else {
+                ctxt.error(builder.getLocation(), "objectFieldOffset string argument must be a literal string");
```

Comment:
@dmlloyd Is it not possible for the field name to not be string literal?

Reply:
No, that will not be supported for Unsafe field accesses. It must be a string literal or traceable back to a string literal (for example a static final field whose value is definitely a string literal).  As a possible exception, we could allow the interpreter to compute a `VmString` as long as it is done at compile time.

Reply:
In that case is it correct to say it is more like a limitation imposed by qbicc on the usage of this API,

Reply:
Yes, but it's not a good idea to use Unsafe generally in most cases, and it's really not a good idea to plug in arbitrary field offsets when doing so.  Most users are going to use `VarHandle` instead.  So, it is a limitation, but not a significant one.